integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -240,6 +241,52 @@ public void shouldModifyProduceMessage(KafkaCluster cluster, Admin admin) throws
         }
     }
 
+    // zero-ack produce requests require special handling because they have no response associated
+    // this checks that Kroxy can handle the basics of forwarding them.
+    @Test
+    public void shouldModifyZeroAckProduceMessage(KafkaCluster cluster, Admin admin) throws Exception {
+        admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+
+        var config = proxy(cluster)
+                .addToFilters(new FilterDefinitionBuilder("ProduceRequestTransformation").withConfig("transformation", TestEncoder.class.getName()).build());
+
+        try (var tester = kroxyliciousTester(config);
+                var producer = tester.producer(Map.of(CLIENT_ID_CONFIG, "shouldModifyProduceMessage", DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000, ACKS_CONFIG, "0"));
+                var consumer = tester
+                        .consumer(Serdes.String(), Serdes.ByteArray(), Map.of(GROUP_ID_CONFIG, "my-group-id", AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+            producer.send(new ProducerRecord<>(TOPIC_1, "my-key", PLAINTEXT)).get();
+            producer.flush();
+
+            ConsumerRecords<String, byte[]> records1;
+            consumer.subscribe(Set.of(TOPIC_1));
+            records1 = consumer.poll(Duration.ofSeconds(10));
+
+            assertEquals(1, records1.count());
+            assertArrayEquals(TOPIC_1_CIPHERTEXT, records1.iterator().next().value());
```

Comment:
We should favour assertThat in new tests.  The most concise incantation I know

```suggestion
            assertThat(records1).hasSize(1);
            assertThat(records1.records(TOPIC_1)).map(ConsumerRecord::value)
                                              .containsExactly(TOPIC_1_CIPHERTEXT);
```